src/index.ts

```diff
@@ -195,11 +197,9 @@ export async function createWithExpiration (peerId: PeerId, value: CID | PeerId
 export async function createWithExpiration (peerId: PeerId, value: CID | PeerId | string, seq: number | bigint, expiration: string, options: CreateOptions = defaultCreateOptions): Promise<IPNSRecord> {
   const expirationDate = NanoDate.fromString(expiration)
   const validityType = IpnsEntry.ValidityType.EOL
+  const lifetimeNs = typeof options.lifetimeNs === "bigint" ? options.lifetimeNs : DEFAULT_TTL
 
-  const ttlMs = expirationDate.toDate().getTime() - Date.now()
-  const ttlNs = (BigInt(ttlMs) * BigInt(100000)) + BigInt(expirationDate.getNano())
```

Comment:
We should probably keep existing code and use this if ttl isnt passed?

Reply:
The existing TTL code doesn't match the suggested [spec](https://specs.ipfs.tech/ipns/ipns-record/#ttl-uint64) is the reason for the code change.  It made it very difficult during the integration process as Kubo is implementing the suggested default TTL per spec.